test/analytics/analytics-on-redshift/lambda/scan-metadata-workflow/store-metadata-into-ddb.test.ts

```diff
@@ -44,151 +44,474 @@ describe('Lambda - store the metadata into DDB from Redshift', () => {
+            { stringValue: 'value1_10#value2_15#value3_25' },
             { stringValue: 'ANDROID#IOS' },
           ],
         ],
       });
-    const dateNowSpy = jest.spyOn(Date, 'now');
-    const timestamp = 1695120657125;
-    dateNowSpy.mockReturnValue(timestamp);
     const resp = await handler(checkScanMetadataStatusEvent);
 
-    expect(dynamoDBMock).toHaveReceivedCommandWith(BatchWriteCommand, {
+
+    expect(dynamoDBMock).toHaveReceivedNthCommandWith(1, BatchGetCommand, {
+      RequestItems: {
+        ClickstreamAnalyticsMetadata: {
+          Keys: [
+            {
+              id: 'projectId#appId#eventName',
+              month: '#202301',
+            },
+          ],
+        },
+      },
+    });
+    expect(dynamoDBMock).toHaveReceivedNthCommandWith(2, BatchGetCommand, {
+      RequestItems: {
+        ClickstreamAnalyticsMetadata: {
+          Keys: [
+            {
+              id: 'projectId#appId#eventName#propertyName#valueType',
+              month: '#202301',
+            },
+          ],
+        },
+      },
+    });
+    expect(dynamoDBMock).toHaveReceivedNthCommandWith(3, BatchGetCommand, {
+      RequestItems: {
+        ClickstreamAnalyticsMetadata: {
+          Keys: [
+            {
+              id: 'projectId#appId#userattributeName#valueType',
+              month: '#202301',
+            },
+          ],
+        },
+      },
+    });
+    expect(dynamoDBMock).toHaveReceivedNthCommandWith(4, BatchWriteCommand, {
       RequestItems: {
         ClickstreamAnalyticsMetadata: [
           {
             PutRequest: {
               Item: {
-                appId: 'appId',
-                createAt: 1695120657125,
-                dataVolumeLastDay: 2000,
-                deleted: false,
-                description: '',
-                displayName: '',
-                hasData: true,
-                id: 'projectId#appId#eventName',
-                metadataSource: 'custom',
-                name: 'eventName',
-                operator: '',
-                platform: ['ANDROID', 'IOS'],
-                prefix: 'EVENT#projectId#appId',
-                projectId: 'projectId',
-                ttl: 1695725457,
-                type: 'EVENT#sprojectId#appId#eventName',
-                updateAt: 1695120657125,
+                ...eventDDBCommonData,
+                day7: {
+                  count: 40,
+                  hasData: true,
+                  name: 'eventName',
+                  platform: ['APP'],
+                },
+                day9: {
+                  count: 40,
+                  hasData: true,
+                  name: 'eventName',
+                  platform: ['ANDROID', 'IOS'],
+                },
+                day11: {
+                  count: 30,
+                  hasData: true,
+                  name: 'eventName',
+                  platform: ['ANDROID', 'WEB'],
+                },
+                summary: {
+                  name: 'eventName',
```

Comment:
why not including  `count` in summary?

Reply:
Calculating the summary when updating the metadata daily would be more efficient.

If the event property does not have any on the current day(said as `day20`), it's OK to use the existing summary and make the field `day20` absent.

Reply:
According to discussion offline, keep current implementation.